### PR TITLE
[PORT] Pipe overlays overhaul and bitmasking

### PR DESCRIPTION
**Pull request by Ghilker**
### Port of hrzntal/horizon#383
Fixes isssues with pipes images on every machines

## Why It's Good For The Game
sprite fix

## Changelog
:cl: @Azarak 
fix: pipes sprite fix
refactor: pipe sprites now uses bitmasks
/:cl:

### CODE REVIEW

**Comment by optimumtact** [comment]
@Kylerace keep an eye on this when it's merged in case it affects map tick

**Comment by Azarak** [comment]
This makes all pipes have 0 overlays instead of 4+, if anything it should help maptick :p

**Comment by optimumtact** [comment]
you'll notice I didn't say in which direction it might affect maptick

**parkour**

**Comment by LemonInTheDark** [comment]
This is based, and I'm excited to see how it maybe effects maptick

**Comment by Kylerace** [comment]
@optimumtact everything appears to just be affecting over/underlays with no vis_contents/added movables, with the relatively low rate i expect these to be updating i see no reason this will make a readable impact in maptick

**Comment by optimumtact** [comment]
/me trips over a block and falls on my face

File: code/modules/atmospherics/machinery/atmosmachinery.dm [hunk 1]
```diff
@@ -426,40 +422,23 @@
...
-		PIPING_LAYER_SHIFT(pipe_overlay, piping_layer)
-		if(trinary == TRUE && (piping_layer == 1 || piping_layer == 5))
-			PIPING_FORWARD_SHIFT(pipe_overlay, piping_layer, 2)
-
-///Similar to getpipeimage(); will create an image from the set_icon and set_state; mostly used to create overlays for connections.
-/obj/machinery/atmospherics/proc/pipe_overlay(set_icon, set_state, direction, color = COLOR_VERY_LIGHT_GRAY, piping_layer = 3, set_layer = PIPE_VISIBLE_LEVEL)
-	var/image/pipe_overlay
-	pipe_overlay = image(icon = set_icon, icon_state = set_state, layer = set_layer, dir = direction)
-	pipe_overlay.color = color
+/obj/machinery/atmospherics/proc/getpipeimage(iconfile, iconstate, direction, col = COLOR_VERY_LIGHT_GRAY, piping_layer = 3, trinary = FALSE)
```
Comment (Kylerace) [inline code/modules/atmospherics/machinery/atmosmachinery.dm:436]: col is iffy, color is better

**Review by Kylerace — CHANGES_REQUESTED** [review]

**Pull request merged**